posthog/models/feature_flag/flag_matching.py

```diff
@@ -364,10 +384,11 @@ def get_highest_priority_match_evaluation(
 
 def hash_key_overrides(team_id: int, person_id: int) -> Dict[str, str]:
     feature_flag_to_key_overrides = {}
-    for feature_flag, override in FeatureFlagHashKeyOverride.objects.filter(
-        person_id=person_id, team=team_id
-    ).values_list("feature_flag_key", "hash_key"):
-        feature_flag_to_key_overrides[feature_flag] = override
+    with execute_with_timeout(FLAG_MATCHING_QUERY_TIMEOUT_MS):
```

Comment:
Is there a reason we don't put this at the top level of the `/decide/` endpoint? Or within the middleware for `/decide/`?

Reply:
We very well _could_, and it would really simplify code a lot.

Reason I haven't done that for now is to not hold on to the connection for that long - since it's all in one transaction, one connection would be in use by decide for the entirety of the request. If we get more than 100 requests at the same instant, that means we will saturate connections completely 😬 .

Vs now, where doing it this way splits out the cpu-bound, and io-bound stuff. (As you can tell by the tests, there are some requests which never even touch the db, because of other optimisation magic, and not giving these requests a connection is awesome)

I _think_ this will be a significant enough optimisation that we should take the code complexity hit, but not sure if you or @ellie disagree?